src/schema/users.ts

```diff
@@ -850,14 +852,56 @@ export const resolvers: IResolvers<any, Context> = {
         .orderBy('date')
         .getRawMany();
     },
-    userStreak: (_, __, ctx: Context, info): Promise<GQLUserStreak> =>
-      graphorm.queryOneOrFail<GQLUserStreak>(ctx, info, (builder) => ({
-        queryBuilder: builder.queryBuilder.where(
-          `"${builder.alias}"."userId" = :id`,
-          { id: ctx.userId },
-        ),
-        ...builder,
-      })),
+    userStreak: async (_, __, ctx: Context, info): Promise<GQLUserStreak> => {
```

Comment:
This logic is quite complex to follow since we're trying to do several things here:
* Workaround our missing data - Should be ignored as once we roll out data should be migrated
* Fetch the streak - This is great but we should do it only in one place, at the beginning of this function. Currently, we do it in different places. Ideally we can add `lastViewAt` as required column in graphorm. Fetch the graphorm in the beginning and do all the checks later.
* Reset the streak - So once we have the streak from the DB, we can check if it requires a reset. Ideally this should be encapsulated in a function. If we need to reset it, we can update the DB without refetching again and just a merged object of the db streak with some adjusted fields.

The above should make the function much more readable imo

Reply:
Cool. Let me refactor it a little bit. Thanks for the points.

Reply:
It looks great now!!